src/internal/nativeModule.web.ts

```diff
@@ -70,12 +70,14 @@ declare global {
     webkitConnection?: Connection;
   }
 }
+const isWindowPresent = typeof window !== 'undefined';
 
-// Check if the browser supports the connection API
-const connection =
-  window.navigator.connection ||
-  window.navigator.mozConnection ||
-  window.navigator.webkitConnection;
+// Check if window exists and if the browser supports the connection API
+const connection = isWindowPresent
+  ? window.navigator.connection ||
+    window.navigator.mozConnection ||
+    window.navigator.webkitConnection
+  : undefined;
```

Comment:
Fascinating that the optional chaining does not work! That's really unexpected. Thanks for trying it, I'm obviously surprised, I would not have suggested it if I thought it would just waste your time :laughing: 

Okay, so we need a specific check, but this still leaves a polymorphic return here of true | false | undefined. I think it should return false at the end?

```suggestion
  : false;
```

?

Reply:
Yeah it's weird, I'm assuming it's how Next is transpiling, or perhaps window *DOES* have a value but it is not an object. I have not checked that.

I am not sure what should be assigned to connection if `window` does not exist 🤷  So fare undefined has allowed me to move forward. let me try with `false`

Reply:
So it works `false` as well, is using `false` more common practice? I can change it to false

Reply:
the original only had `Connection | undefined` 
![image](https://user-images.githubusercontent.com/44002321/204318365-de58eae4-1a42-483e-a3d6-45da2b6d2585.png)

Reply:
Oh I see - I am sorry, I misread the conditional chain before and thought it was returning true | false as I was commenting earlier, that's my mistake. If it was returning Connection | undefined, then my suggestion is the opposite of my intention. My intention was to follow the best practice of not altering return types in what would be a patch fix (or ever - if not necessary). This looks fine then, sorry for the churn - you've got something tested + working with identical return types to previous, that sounds great